support-frontend/assets/helpers/redux/checkout/payment/contributionsSideEffects.ts

```diff
@@ -20,17 +21,17 @@ import {
 export function addPaymentsSideEffects(
 	startListening: ContributionsStartListening,
 ): void {
-	startListening({
-		actionCreator: setPopupOpen,
-		effect() {
-			// TODO: we should do this on the payment method selection action instead in future
-			storage.setSession('selectedPaymentMethod', DirectDebit);
-		},
-	});
-
 	startListening({
 		matcher: shouldCheckFormEnabled,
-		effect(_action, listenerApi) {
+		effect(action, listenerApi) {
+			if (setPaymentMethod.match(action)) {
+				const paymentMethod = action.payload;
+				const { payPal } = listenerApi.getState().page.checkoutForm.payment;
+				storage.setSession('selectedPaymentMethod', paymentMethod);
+				if (paymentMethod === PayPal && !payPal.hasBegunLoading) {
+					void listenerApi.dispatch(loadPayPalExpressSdk());
+				}
+			}
```

Comment:
could this be in a separate call to `startListening`? eg


```ts
startListening({
		action: setPaymentMethod,
		effect(action, listenerApi) {
				const paymentMethod = action.payload;
				const { payPal } = listenerApi.getState().page.checkoutForm.payment;
				storage.setSession('selectedPaymentMethod', paymentMethod);
				if (paymentMethod === PayPal && !payPal.hasBegunLoading) {
					void listenerApi.dispatch(loadPayPalExpressSdk());
				}
			}});
```

Reply:
I think it's preferable to limit the amount of listeners we create, especially ones that will respond to the same actions, but it might be better to pull these effects out into separate functions.